mne/filter.py

```diff
@@ -2498,6 +2499,7 @@ def savgol_filter(self, h_freq, verbose=None):
         from scipy.signal import savgol_filter
 
         _check_preload(self, "inst.savgol_filter")
+        _check_empty(self, "savgol_filter")
```

Comment:
... and then the common `_check_preload() _check_empty()` pattern makes me think we should have `_check_preload` *always* just check for emptiness, too. That will probably satisfy 90-100% of our internal uses of `_check_preload` for epochs and if there are 10% where we *do* want to allow preload-and-no-epochs then we can add `_check_preload(..., allow_empty=False)` kwarg and pass `allow_empty=True` to the ones that need it.